crowdfunding/users/serializers.py

```diff
@@ -0,0 +1,122 @@
+from rest_framework import serializers
+from .models import CustomUser
+from django.contrib.auth.models import User
+from rest_framework.validators import UniqueValidator
+from django.contrib.auth.password_validation import validate_password
+
+
+# Serializer to Get User Details using Django Token Authentication
+class CustomUserSerializer(serializers.ModelSerializer):
+    username = serializers.CharField(max_length=200)
+    first_name = serializers.CharField(max_length=200)
+    last_name = serializers.CharField(max_length=200)
+    email = serializers.EmailField(required=True,
+                                   validators=[UniqueValidator(queryset=CustomUser.objects.all())])
+    password = serializers.CharField(
+        write_only=True, required=True, validators=[validate_password])
+    password2 = serializers.CharField(write_only=True, required=True)
+
+    class Meta:
+        model = CustomUser
+        fields = ['username', 'first_name', 'last_name',
+                  'email', 'password', 'password2']
+        extra_kwargs = {'first_name': {'required': True},
+                        'last_name': {'required': True}}
+
+    def create(self, validated_data):
+        user = CustomUser.objects.create(
+            username=validated_data['username'],
+            email=validated_data['email'],
+            first_name=validated_data['first_name'],
+            last_name=validated_data['last_name']
+        )
+        user.set_password(validated_data['password'])
+        user.save()
+        return user
+
+    def validate(self, attrs):
+        if attrs['password'] != attrs['password2']:
+            raise serializers.ValidationError(
+                {"password": "Your password fields didn't match."})
+        return attrs
+
+
+class CustomUserDetailSerializer(CustomUserSerializer):
+    class Meta:
+        model = CustomUser
+        fields = ['username', 'first_name', 'last_name',
+                  'email', 'password', 'password2']
+        extra_kwargs = {'first_name': {'required': True},
+                        'last_name': {'required': True}}
+
+
+# Serializer to Register User
+# class CustomUserRegisterSerializer(serializers.ModelSerializer):
+#     class Meta:
+#         model = CustomUser
+
+
+class CustomUserChangePasswordSerializer(serializers.ModelSerializer):
+    password = serializers.CharField(
+        write_only=True, required=True, validators=[validate_password])
+    password2 = serializers.CharField(write_only=True, required=True)
+    old_password = serializers.CharField(write_only=True, required=True)
```

Comment:
you may not need password2 here, as it's probably the job of the front end to check input